### PR TITLE
Fix Integration Issues Noticed in NetFixClient

### PR DESCRIPTION
**Pull request by Brett208**
Apparently marking Checksum as const is not binary compatible. Also missed changing a forward declare.

### CODE REVIEW

File: Network/Packet.h [hunk 1]
```diff
@@ -232,7 +232,7 @@ namespace OP2Internal
 		};
 
 		// Member functions
-		int Checksum() const;
+		int Checksum();
```
Comment (DanRStevens) [inline Network/Packet.h]: Could you elaborate on the problem here?

The `const` might affect the name mangling, which will affect linking. The decorated symbol name may need to be updated in the .asm file. The expected decorated name will be given in an error message during linking.

OP2Internal.asm: line 492:
```asm
Export 0x00490F10, ?Checksum@Packet@OP2Internal@@QAEHXZ
```

I double checked the function at the listed address `0x00490F10` in Outpost2.exe. It is indeed `const`. It makes not attempt to modify the `this` object. We should try to maintain `const` here, which likely requires updating the decorated symbol name.

**Review by DanRStevens — CHANGES_REQUESTED** [review]
Good catch on the forward declare I missed updating.

----

I think you were right in your initial observations that `Checksum()` should be marked `const`. I believe we should try to maintain that.

**Comment by DanRStevens** [comment]
I checked the AppVeyor errors from the other branch, and pushed a change with the name mangling update which maintains the `const` on `Checksum`. I think this is the change we want to merge here.

**Comment by Brett208** [comment]
Looks good. Thanks for fixing the name mangling issue. 

How are you checking if the original function is const? Using ollydbg?

Thanks,
Brett

**Comment by DanRStevens** [comment]
Yes, OllyDbg. The method is only about 2 pages of code. It's quite quick to scan it for memory references as the left side operand (i.e. writes). There's only a few, and they all write to the stack. Hence, only function local variables are written, not object member variables.

**Pull request merged**